backend/controller/user.go

```diff
@@ -2,14 +2,87 @@ package controller
 
 import (
 	"api/config"
+	"api/helpers"
 	"api/models"
 	"encoding/json"
+	"github.com/go-chi/chi/v5"
+	"log"
 	"net/http"
+	"strconv"
 )
 
 func GetAllUsers(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
 	users := []models.User{}
-	config.DB.Find(&users)
+	result := config.DB.Find(&users)
+	if result.Error != nil {
+		helpers.DBErrorHandling(result.Error, w, r)
+		return
+	}
+	//go yells at you if you don't error check this, better to be safe
+	errJson := json.NewEncoder(w).Encode(&users)
+	if errJson != nil {
+		http.Error(w, "500", http.StatusInternalServerError)
+		return
+	}
+}
+func GetUser(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	uId := chi.URLParam(r, "userId")
+	var user models.User
+	result := config.DB.First(&user, uId)
+
+	if result.Error != nil {
+		helpers.DBErrorHandling(result.Error, w, r)
+		return
+	}
+	errJson := json.NewEncoder(w).Encode(&user)
+	if errJson != nil {
+		http.Error(w, "500", http.StatusInternalServerError)
+		return
+	}
+}
+
+func CreateUser(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	var user models.UserReq
+	err := json.NewDecoder(r.Body).Decode(&user)
+	if err != nil || !helpers.CheckFields(user) {
+		if err != nil {
+			log.Print(err)
+		} else {
+			log.Print("Request doesn't contain all params")
+		}
+		http.Error(w, "400", http.StatusBadRequest)
+		return
+	}
+	dbUser := models.User{Name: user.Name,
+		Email: user.Email, IsAdmin: user.IsAdmin}
+	result := config.DB.Create(&dbUser)
+	if result.Error != nil {
+		helpers.DBErrorHandling(result.Error, w, r)
+		return
+	}
+	errJson := json.NewEncoder(w).Encode(r.Host + "/api/users/{" + strconv.Itoa(int(dbUser.ID)) + "}")
```

Comment:
What does this do?

Reply:
we could if thats the wanted functionality

Reply:
depends on the intended use case, I can update it as you think you need

Reply:
This just seems unnecessary steps to me, ill let cameron decide what to do

Reply:
he said he'll look into it

Reply:
Apparently the best practice for this is to return the newly created/updated object in the same format you would for a GET request for that object.

Reply:
But can we achieve this without kicking off another request? I've never seen this pattern of actually calling a GET? Like wouldn't even just calling `GetAllUsers` be better?

Reply:
To clarify, I am not talking about kicking off another request, I'm talking about avoiding one.
For example:
POST `/user` creates user with ID 123. Creating this user (the response of the POST request) would return the same response as GET `/user/123`